tests/managers/screen/choiceset/ChoiceSetManagerTests.js

```diff
@@ -0,0 +1,211 @@
+
+            Validator.assertEquals(returnedChoices.length, 1);
+            for (const cell of returnedChoices) {
+                Validator.assertEquals(cell.getText(), 'test2');
+            }
+        });
+
+        it('testPresentingKeyboardShouldReturnCancelIDIfKeyboardCanBeSent', function () {
+            const cancelId = csm.presentKeyboard('initial text', null, new SDL.manager.screen.choiceset.KeyboardListener());
+            Validator.assertNotNull(cancelId);
+        });
+
+        it('testPresentingKeyboardShouldNotReturnCancelIDIfKeyboardCannotBeSent', function () {
+            const oldState = csm._state;
+            csm._state = SDL.manager._SubManagerBase.ERROR;
+            const cancelId = csm.presentKeyboard('initial text', null, new SDL.manager.screen.choiceset.KeyboardListener());
+            Validator.assertNull(cancelId);
+            csm._state = oldState;
+        });
+
+        it('testDismissingExecutingKeyboard', function () {
+            const testCancelId = 42;
+            const testKeyboardOp = new SDL.manager.screen.choiceset._PresentKeyboardOperation(sdlManager._lifecycleManager);
+            testKeyboardOp._cancelId = testCancelId;
+
+            const callback = sinon.fake(() => {});
+            const stub = sinon.stub(testKeyboardOp, '_dismissKeyboard')
+                .callsFake(callback);
+
+            csm._currentlyPresentedKeyboardOperation = testKeyboardOp;
+            csm.dismissKeyboard(testCancelId);
+            Validator.assertEquals(callback.called, true);
+            stub.restore();
+        });
+
+        it('testDismissingQueuedKeyboard', function () {
+            const testCancelId = 42;
+
+            // Currently executing operation
+            const testKeyboardOp = new SDL.manager.screen.choiceset._PresentKeyboardOperation(sdlManager._lifecycleManager);
+            testKeyboardOp._cancelId = 96;
+            csm._currentlyPresentedKeyboardOperation = testKeyboardOp;
+
+            // Queued operations
+            const testKeyboardOp2 = new SDL.manager.screen.choiceset._PresentKeyboardOperation(sdlManager._lifecycleManager);
+            testKeyboardOp2._cancelId = 42;
+            csm._currentlyPresentedKeyboardOperation = testKeyboardOp2;
+
+            const callback = sinon.fake(() => {});
+            const stub = sinon.stub(testKeyboardOp, '_dismissKeyboard')
+                .callsFake(callback);
+
+            const callback2 = sinon.fake(() => {});
+            const stub2 = sinon.stub(testKeyboardOp2, '_dismissKeyboard')
+                .callsFake(callback2);
+
+            // Queued operation should be canceled
+            csm.dismissKeyboard(testCancelId);
+            Validator.assertEquals(callback.called, false);
+            Validator.assertEquals(callback2.called, true);
+            stub.restore();
+            stub2.restore();
+        });
+
+        it('testAddUniqueNamesToCells', function () {
```

Comment:
This method has some linter errors relating to double quotes for strings, trailing spaces, and padding from blank lines.

Reply:
Addressed